src/stream/src/executor/sort_buffer.rs

```diff
@@ -140,36 +140,25 @@ impl<S: StateStore> SortBuffer<S> {
             // Only records with timestamp greater than the last watermark will be output, so
             // records will only be emitted exactly once unless recovery.
             let start_bound = if let Some(last_watermark) = last_watermark.clone() {
-                // TODO: `start_bound` is wrong here, only values with `val.0 > last_watermark`
-                // should be output, but it's hard to represent `OwnedRow::MAX`. A possible
-                // implementation is introducing `next_unit` on a subset of `ScalarImpl` variants.
-                // Currently, we can skip some values explicitly.
-                Bound::Excluded((last_watermark, OwnedRow::empty().into()))
+                Bound::Excluded((
+                    // TODO: unsupported type or watermark overflow. Do we have better ways instead
+                    // of unwrap?
+                    last_watermark.successor().unwrap(),
+                    OwnedRow::empty().into(),
+                ))
```

Comment:
I think it is ok and the optimizer should ensure that. Also, maybe a better error message here is better.

Reply:
Or do we need to use `Unbounded` when overflowing (corner case, never mind 😄)?

Reply:
We can discuss them later